pkg/reconciler/vspheresource/resources/names/names.go

```diff
@@ -28,5 +28,8 @@ func RoleBinding(vms *v1alpha1.VSphereSource) string {
 }
 
 func ServiceAccount(vms *v1alpha1.VSphereSource) string {
-	return kmeta.ChildName(vms.Name, "-serviceaccount")
+	if vms.Spec.ServiceAccountName == "" {
+		return kmeta.ChildName(vms.Name, "-serviceaccount")
```

Comment:
I think this prevents the pod from ever using the default service account since an empty field is converted to `sourcename-serviceaccount`. We should no longer be creating a new service account per source if I understand correctly. 

This function can likely go away now that service account can either be "" or a custom field set by user.